qa/tasks/notification_tests.py

```diff
@@ -299,7 +299,8 @@ def task(ctx,config):
                     'port':endpoint.port,
                     'host':endpoint.dns_name,
                     'zonegroup':'default',
-                    'cluster':'noname'
+                    'cluster':'noname',
```

Comment:
@yuvalif with `notification_v2` now on by default in the code, this setup is not workin. I mean by default the vstart or test-rgw-multisite setup will always have v2 enabled.
so we need to change the logic here and in `init.py` to disable the v2 if we want to test the v1.
also note now with `vstart` we cannot test `v1` coz we cannot disable v2 without restarting the gateway :(

Reply:
@cbodley if we use v2 by default in the code, how would we change to v1 before doing an upgrade?

Reply:
it looks like you have it right in the test: `zonegroup modify --disable-feature=`

Reply:
> it looks like you have it right in the test: `zonegroup modify --disable-feature=`

i have 2 issues with it:
* i will have to call that before the upgrade, is it possible to disable the feature before we have the new code?
* what about cases where we don't have multisite and there is no realm defined. do we have to restart all RGWs manually before the upgrade?

Reply:
can you clarify whether we're really talking about "upgrade" as in ceph reef -> squid? or are we just talking about testing the migration from squid v1 -> squid v2?

> i will have to call that before the upgrade, is it possible to disable the feature before we have the new code?

for the upgrade case there will be an existing zone and zonegroup, created in reef or before, that doesn't list the new feature as supported or enabled. upgrading to squid won't enable the feature automatically, you'd have to do that manually. it's only enabled by default if you create a new zone/zonegroup using the squid release

> what about cases where we don't have multisite and there is no realm defined. do we have to restart all RGWs manually before the upgrade?

changes to zone/zonegroup config either require a period commit (for realms) or radosgw restarts, right. if a squid test case is creating the zone/zonegroup for migration testing, you can pass the `--disable-feature=` option to the zonegroup/zone create commands to avoid the need for restarts

in the rgw/multisite teuthology suite, zone/zonegroup configuration is managed by the [rgw_multisite.py task](https://github.com/ceph/ceph/blob/8be184a6/qa/tasks/rgw_multisite.py) which expects a bunch of yaml as input (ex. [realms/three-zones.yaml](https://github.com/ceph/ceph/blob/8be184a6/qa/suites/rgw/multisite/realms/three-zones.yaml#L11)). that task creates zonegroups by passing their json description into `radosgw-admin zonegroup set`. that json description comes directly from the yaml, so you have full control of the enabled features: `enabled_features: ['resharding']`

Reply:
> can you clarify whether we're really talking about "upgrade" as in ceph reef -> squid? or are we just talking about testing the migration from squid v1 -> squid v2?

reef/quincy -> squid

> upgrading to squid won't enable the feature automatically, you'd have to do that manually.

ok. thanks for clarifying

>  it's only enabled by default if you create a new zone/zonegroup using the squid release

the definition is per zonegroup. creating a new zone in a v1/reef created zonegroup should not enable it

> you can pass the --disable-feature= option to the zonegroup/zone create commands to avoid the need for restarts

we should probably document that adding a new zonegroup may reequire using `--disable-feature=`
for the following case:
* upgrade reef->squid
* add new zonegroup before the upgrade is done (there are still reef RGWs running)